src/components/LayoutGroup.tsx

```diff
@@ -18,6 +18,7 @@ export interface LayoutGroupProps {
 export function LayoutGroup({ children }: LayoutGroupProps) {
   const createQueue = useRef(new Set<() => void>()).current;
   const destroyQueue = useRef(new Set<() => void>()).current;
+  const isMounted = useRef(true);
```

Comment:
Is this actually true, initially? 😆

Reply:
I meeeaaannnn... I guess it depends on your definition of mounted haha. Given the way that this is being used, I suppose it's safe to set this to false initially and set it true in a useLayoutEffect creator.

Reply:
If so, I'm tempted by the idea of not queueing child destroys, but providing a simpler explanation for why.

Reply:
It's not, I've checked. It is child->parent in regular updates, and actually behaves differently and runs parent->child in during unmounts!

Reply:
That's true even if both the parent and the child have effects with dependencies that changed and the render was caused by a state update at the parent or above?

Reply:
I was testing with effects with no dependencies because it seemed easier to coordinate. But yes, if you use effects with no dependencies, and update the state at the parent, effect cleanups are run from child -> parent. If you unmount the parent, they are run from parent -> child

Reply:
That's super wild.

Reply:
For anyone that wants to follow along, here's a [minimal reproduction on CodeSandbox](https://codesandbox.io/p/sandbox/goofy-brook-tyc87m?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clsz0h67t00062e69ppema5fd%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clsz0h67t00022e696gdauoqe%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clsz0h67t00032e69d0dlryqu%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clsz0h67t00052e69c281fr1z%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clsz0h67t00022e696gdauoqe%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsz0h67t00012e69sqat2v6j%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clsz0ieva00022e692uqb170o%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clsz0h67t00022e696gdauoqe%2522%252C%2522activeTabId%2522%253A%2522clsz0ieva00022e692uqb170o%2522%257D%252C%2522clsz0h67t00052e69c281fr1z%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clsz0h67t00042e69s4in6ui7%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clsz0h67t00052e69c281fr1z%2522%252C%2522activeTabId%2522%253A%2522clsz0h67t00042e69s4in6ui7%2522%257D%252C%2522clsz0h67t00032e69d0dlryqu%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clsz0h67t00032e69d0dlryqu%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D).

If you open the console, you'll see that the order of the print statements changes depending on whether you click Update or Toggle mount!